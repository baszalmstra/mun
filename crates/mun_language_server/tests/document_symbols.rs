mod support;

#[test]
fn test_document_symbols() {
    let mut server = support::Project::with_fixture(
        r#"
    //- /Mun.toml
    [package]
    name = "foo"
    version = "0.0.0"

    //- /src/mod.mun
    fn main() -> i32 {}

    struct Foo {}

    type Bar = Foo;
    "#,
    )
    .server();

    let response = async_std::task::block_on(
        server.send_request::<lsp_types::request::DocumentSymbolRequest>(
            lsp_types::DocumentSymbolParams {
                text_document: server.doc_id("src/mod.mun"),
                work_done_progress_params: Default::default(),
                partial_result_params: Default::default(),
            },
        ),
    );

    insta::assert_debug_snapshot!(response);
}
