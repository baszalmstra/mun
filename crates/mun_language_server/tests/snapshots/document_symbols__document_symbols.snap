---
source: crates/mun_language_server/tests/document_symbols.rs
expression: response
---
Some(
    Nested(
        [
            DocumentSymbol {
                name: "main",
                detail: Some(
                    "fn() -> i32",
                ),
                kind: Function,
                deprecated: None,
                range: Range {
                    start: Position {
                        line: 0,
                        character: 0,
                    },
                    end: Position {
                        line: 0,
                        character: 19,
                    },
                },
                selection_range: Range {
                    start: Position {
                        line: 0,
                        character: 3,
                    },
                    end: Position {
                        line: 0,
                        character: 7,
                    },
                },
                children: None,
            },
            DocumentSymbol {
                name: "Foo",
                detail: None,
                kind: Struct,
                deprecated: None,
                range: Range {
                    start: Position {
                        line: 2,
                        character: 0,
                    },
                    end: Position {
                        line: 2,
                        character: 13,
                    },
                },
                selection_range: Range {
                    start: Position {
                        line: 2,
                        character: 7,
                    },
                    end: Position {
                        line: 2,
                        character: 10,
                    },
                },
                children: None,
            },
            DocumentSymbol {
                name: "Bar",
                detail: Some(
                    "Foo",
                ),
                kind: TypeParameter,
                deprecated: None,
                range: Range {
                    start: Position {
                        line: 4,
                        character: 0,
                    },
                    end: Position {
                        line: 4,
                        character: 15,
                    },
                },
                selection_range: Range {
                    start: Position {
                        line: 4,
                        character: 5,
                    },
                    end: Position {
                        line: 4,
                        character: 8,
                    },
                },
                children: None,
            },
        ],
    ),
)
